### PR TITLE
GEODE-5438: Add concurrency test helper to run list of tasks

### PR DESCRIPTION
**Pull request by balesh2**
Signed-off-by: Helena A. Bales <hbales@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelper.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+   *         input. If there are no exceptions thrown in the threads then the list of results will
+   *         be in the
+   *         order of the callables passed in. If there are exceptions, the stacktrace of the
+   *         exceptions
+   *         will be printed and a RuntimeException will be thrown.
+   * @throws RuntimeException if any of the threads throw ExecutionException or
+   *         CancellationException
+   */
+  public static <T> List<T> runCallables(List<Callable<T>> callables) throws InterruptedException {
+    return runCallables(callables, 30);
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelper.java]: Is the default supposed to be 30 seconds? javadoc says 300.

File: geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelperTest.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+  @Test
+  public void testCallablesWithDefaultTimeoutWithNoExceptions() {
+    List<Callable<Boolean>> callables = Arrays.asList(c1, c4);
+    List<Boolean> results = null;
+
+    try {
+      results = ConcurrencyTestHelper.runCallables(callables);
+    } catch (Exception e) {
+      fail("Expected no exceptions to be thrown.");
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelperTest.java]: The expectations for the Callable tests with no expected exceptions it's not really necessary to do the `runCallables()` in a try/catch. Just add a throws declaration to the test methods and let any exceptions propagate up to the test runner where they will cause a test failure.

File: geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+   *         input. If there are no exceptions thrown in the threads then the list of results will
+   *         be in the
+   *         order of the callables passed in. If there are exceptions, the stacktrace of the
+   *         exceptions
+   *         will be printed and a RuntimeException will be thrown.
+   * @throws RuntimeException if any of the threads throw ExecutionException or
+   *         CancellationException
+   */
+  public static <T> List<T> runCallables(List<Callable<T>> callables, Integer timeoutSeconds)
+      throws InterruptedException {
```
Comment (pdxrunner) [inline geode-core/src/test/java/org/apache/geode/management/ConcurrencyTestHelper.java]: Would it be useful to return the results of the Callables even if one or more of them threw an exception? Also, the list of exceptions should identify which callable the exception is associated with. In the IntegrationTest included with this PR, the tests that have more than one callable throwing an exeception don't check if the excpetions came from the expected callables.

**Comment by balesh2** [comment]
We will be making significant changes to this implementation to provide a better API. I will send a proposal for the API to the geode dev list. Closing this PR for now.

**Pull request closed without merging**